action_create.go

```diff
@@ -0,0 +1,80 @@
+package usermngmt
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/Selly-Modules/logger"
+	"github.com/Selly-Modules/mongodb"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+)
+
+// CreateOptions ...
+type CreateOptions struct {
+	Name           string
+	Phone          string
+	Email          string
+	HashedPassword string
+	Status         string
+	RoleID         primitive.ObjectID
+	Other          string
+}
+
+// Create ...
+func (s Service) Create(payload CreateOptions) error {
+	var (
+		col = s.getUserCollection()
+		ctx = context.Background()
+	)
+
+	// Validate payload
+	err := payload.validate()
+	if err != nil {
+		return err
+	}
+
+	// New user data from payload
+	userData, err := payload.newUser()
+	if err != nil {
+		return err
+	}
+
+	//  Find roleID exists or not
+	if !s.isRoleIDAlreadyExisted(ctx, userData.RoleID) {
+		return errors.New("roleID does not exist")
```

Comment:
roleID does not exist -> role id does not exist

lỗi là ngữ cảnh cho user đọc, nên viết clear tí

Reply:
done